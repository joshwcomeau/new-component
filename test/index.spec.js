var spawnCommand = require('spawn-command');

const CLI_PATH = require.resolve('../src/index');

describe('new-component CLI', () => {
  it('should accept --help argument', done => {
    child = spawnCommand(CLI_PATH + ' --help');

    child.stdout.on('data', function (data) {
      const output = data.toString();
      expect(output).toMatchSnapshot();
      expect(output).toContain('-t, --type <componentType>');
    });

    child.on('exit', function (exitCode) {
      expect(exitCode).toEqual(0);
      done(); // must always call on exit to test async code
    });
  })
});
