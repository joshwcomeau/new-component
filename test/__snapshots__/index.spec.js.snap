// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should accept --help argument 1`] = `
"
  Usage: index [options] <componentName>

  Options:

    -V, --version                    output the version number
    -t, --type <componentType>       Type of React component to generate (default: \\"class\\") (default: class)
    -d, --dir <pathToDirectory>      Path to the \\"components\\" directory (default: \\"src/components\\") (default: src/components)
    -x, --extension <fileExtension>  Which file extension to use for the component (default: \\"js\\") (default: js)
    -h, --help                       output usage information
"
`;
